table/tables/partition.go

```diff
@@ -91,8 +91,11 @@ func (t *partitionedTable) GetPartitionedTable() table.PartitionedTable {
 // partitionedTable is a table, it contains many Partitions.
 type partitionedTable struct {
 	TableCommon
-	partitionExpr   *PartitionExpr
-	partitions      map[int64]*partition
+	partitionExpr *PartitionExpr
+	partitions    map[int64]*partition
+	// originalPartitionExpr is only used by truncating table partition.
+	originalPartitionExpr *PartitionExpr
```

Comment:
Why is this needed, just because the number of partitions have changed?

Reply:
This data member is removed in the new PR.